test.ts

```diff
@@ -71,4 +71,75 @@ describe.each([
       expect(splitLines(result)).toStrictEqual(splitLines(expected))
     }
   )
+
+  test.each([
+    'GCN 123',
+    'GCN #123',
+    '(GCN 123)',
+    '(GCN #123)',
+    'GCN circ. 123',
+    'GCN circ. #123',
+    '(GCN circ. #123)',
+    'GCN circ 123',
+    '(GCN circ 123)',
+    'GCN Circ. 123',
+    'GCN Circ. #123',
+    '(GCN Circ. #123)',
+    'GCN Circ 123',
+    '(GCN Circ 123)',
+    'GCN Circular 123',
+    'GCN Circular #123',
+    '(GCN Circular 123)',
+    '(GCN Circular #123)',
+    'GCN Circs 123, 456',
+    'GCN Circs. 123, 456',
+    'GCN Circs #123, #456',
+    '(GCN Circs #123, #456)',
+    '(GCN Circs 123, 456)',
+    'GCN Circ 123, 456',
+    'GCN Circ. 123, 456',
+    'GCN Circ #123, #456',
+    '(GCN Circ 123, 456)',
+    'GCN circs 123, 456',
+    'GCN circs. 123, 456',
+    'GCN circs #123, #456',
+    '(GCN circs 123, 456)',
+    'GCN circs 123 and 456',
+    'GCN circs. 123 and 456',
+    'GCN circs #123 and #456',
+    '(GCN circs 123 and 456)',
+    'GCN circs 123 456',
+    'GCN circs. 123 456',
+    'GCN circs #123 #456',
+    '(GCN circs 123 456)',
+    'GCN          #123',
+    'gcn         123',
```

Comment:
Instead of creating a new test, please add these examples to the `test.md` file and use the existing test infrastructure. If you delete the `test.json` and `test.html` files, then running the tests will generate `test.draft.json` and `test.draft.html` files which you can review and then move into place.

This will be even easier with #45 because that PR will allow you to do that by running `npm test -- --updateSnapshot`.

Reply:
okay, but then anyone that wants to see a list of accepted patterns will have to go digging for them instead of having a convenient list.

Reply:
You find the sample file, https://github.com/nasa-gcn/remark-rehype-astro/blob/main/src/replacements/gcn-circular/test.md, inconvenient?

Reply:
Personal preference wise, yes. Neither way is "right" or "wrong," they are just different options. Some people prefer having less test cases and just doing it all in one massive test because it is more minimalist looking. Some people like the granularity of small test cases and small test objects and not having to hunt all over for an individual test case.

I personally love it when people have each case listed clearly in the test and I don't have to hunt down another file or pick through anything to see what the case under test is. That way if you are developing or making documentation about what is acceptable, or just trying to make sure you know yourself, the list is just right there. You don't have to dig through 100 lines of babble and pick out what the patterns are from that. 

Additionally I think it's much more challenging to go over a massive response file (because it's *quite* a lot larger now) and proofread to make sure there are no errors. I don't particularly like leaving any part of testing up to a human because we can miss things, especially in big files, which means our tests are only as strong as our mock response. If we miss something in the response file, our test won't catch the issue whereas the way I had it originally would catch it because it's just testing against the expected pattern not a predetermined response. When the cases are run individually with small test objects, I feel like there is less room for error than when you're manually checking a 1542 line json object.

Reply:
I see what you mean. I went with the response files originally because I envisioned that they would also serve as a quick reference of supported syntax for end users who are comfortable reading Markdown but not Jest test cases.

Reply:
For example, start a timer, then open the test.md I just uploaded, and collect a list of the patterns under test and post them in a comment. How long does it take to gather that? If we use a list of patterns under test instead, how long does it take you to post a list of the patterns? As long as it takes to cut and paste.

Reply:
I don't understand your last comment. Please elaborate.

Reply:
Would you please remove the custom test code for now? I'm open to changing the design of the test case and I would like to hear more, but in a separate issue or PR.

Reply:
if the patterns buried within a varying patterns of text (which is the strength doing it in one file, you can change the preceding and following test and include test cases where it shouldn't match) then you would have to pick each case out instead of just cutting and pasting a list. That is how I originally changed it after you requested it be in the test.md file because it was a more realistic test case and I could also test non-matches. 

By changing it to be a list of patterns in the test file as you subsequently requested, it makes it behave basically like a list of patterns like I had before (but I did add preceding and following text to make it more realistic so it's different there), which removes the added testing potential for non-matches in the same sentence as the real match, but does make it more clear.

Reply:
I did

Reply:
I don't think you did. It's still here in the diff.